force-app/main/default/classes/DMLOperations.cls

```diff
@@ -146,10 +177,60 @@ public with sharing class DMLOperations {
      * @param oppNames The list of Opportunity names.
      */
     public static void upsertOpportunities(String accountName, List<String> oppNames) {
-        // Loop through each Opportunity name
-            // Create an Opportunity with the given name
-            // Add the Opportunity to the list
-    }
+        
+    Id myAccountId;
+            
+        List<Account> myNeededAcctIds = [SELECT Id FROM Account WHERE Name = :accountName WITH USER_MODE LIMIT 1];
+            if(myNeededAcctIds.isEmpty()){
+                Account account = new Account(Name = accountName); 
+                    Database.insert(account, AccessLevel.USER_MODE);
+                myAccountId = account.Id;
+                system.debug('New Acct Id: ' + myAccountId);
+                }
+        
+                else {
+                    myAccountId = myNeededAcctIds.get(0).Id;
+                }
+            List<Opportunity> existingOpps = [SELECT Id, Name FROM Opportunity WHERE Name IN :OppNames];      
+        
+                Set<String> existingOppNames = new Set<String>();
+                    for (Opportunity eachOpp : existingOpps){
+                    existingOppNames.add(eachOpp.Name);
+                    }
+        
+            List<Opportunity> oppsToUpsert = new List<Opportunity>();    
+            
+            
+            
+        for (String OppName : OppNames) {
+            if(existingOppNames.contains(OppName)){
+                
+                for (Opportunity eachOpp : existingOpps) {
+                if (eachOpp.Name == oppName) {
+                    eachOpp.AccountId = myAccountId;// Opportunity exists, associate with Account and add to upsert list
+                    oppsToUpsert.add(eachOpp);
+                }
+            }
+        } else {
+            Opportunity opp = new Opportunity();
+                opp.Name = oppName;
+                opp.StageName = 'Qualification';
+                opp.CloseDate = Date.Today().addMonths(3);
+                opp.Amount = 50000;
+                opp.AccountID = myAccountId;
+                
+                oppsToUpsert.add(opp);
+            System.debug('New Opp to upsert?' + opp);
```

Comment:
This method seems very long.
Try to do it in these steps.

1.  Query the opps
2. Check if the opps list is more than 0
3. If it is, loop through those and add the accountId
4. Else loop through all other opps and create a new Opportunity 
5. Add the new opps to the list that was querried
6. Upsert